src/lightning_lite/utilities/apply_func.py

```diff
@@ -105,9 +105,4 @@ def convert_to_tensors(data: Any, device: _DEVICE) -> Any:
     # convert non-tensors
     for src_dtype, conversion_func in CONVERSION_DTYPES:
         data = apply_to_collection(data, src_dtype, conversion_func, device=device)
-
-    def _move_to_device_and_make_contiguous(t: Tensor, device: _DEVICE) -> Tensor:
-        return t.to(device).contiguous()  # type: ignore[arg-type]
```

Comment:
With the changes in this PR, contiguous shouldn't be necessary anymore as it was added to the distributed calls themselves.

This is technically a breaking change, but it should close the gap between pure-pytorch and pytorch-lightning.

Reply:
In which version was it introduced? Don't we need to keep it for the older PyTorch versions we support?

Reply:
I wrote my comment above before realizing that they didn't properly fix it. It was supposed to be introduced in 1.12

So we still have this `contiguous()` call, but it's now moved to https://github.com/Lightning-AI/lightning/pull/11531/files#diff-b49e7b638737b063bc51ee0cc6727c82a3ff52bf72a40f6a49292818b2834910R176 (instead of in `convert_to_tensors`) which is the only place that needs it to the best of my knowledge